File: behavioral/python/mediator/Mediator.py
from Telephone import Telephone
from Computer  import Computer
from Tablet    import Tablet

class Mediator:
    __telephone: Telephone = None
    __computer:  Computer  = None
    __tablet:    Tablet    = None

    def setTelephone(self, telephone: Telephone) ->None:
        self.__telephone = telephone

    def getTelephone(self) ->Telephone:
        return self.__telephone

    def setComputer(self, computer: Computer) ->None:
        self.__computer = computer

    def getComputer(self) ->Computer:
        return self.__computer

    def setTablet(self, tablet: Tablet) ->None:
        self.__tablet = tablet

    def getTablet(self) ->Tablet:
        return self.__tablet

    def runTelephone(self) ->None:
        self.__computer.endTest()
        self.__tablet.endTest()

    def runTablet(self) ->None:
        self.__computer.endTest()
        self.__telephone.endTest()

    def runComputer(self) ->None:
        self.__telephone.endTest()
        self.__tablet.endTest()

File: behavioral/Java/mediator/src/Mediator.java
public class Mediator {
    private Telephone telephone;
    private Tablet    tablet;
    private Computer  computer;

    public Telephone getTelephone() {
        return telephone;
    }

    public void setTelephone(Telephone telephone) {
        this.telephone = telephone;
    }

    public Tablet getTablet() {
        return tablet;
    }

    public void setTablet(Tablet tablet) {
        this.tablet = tablet;
    }

    public Computer getComputer() {
        return computer;
    }

    public void setComputer(Computer computer) {
        this.computer = computer;
    }

    public void runTelephone(){
        this.computer.endTest();
        this.tablet.endTest();
    }

    public void runTablet(){
        this.computer.endTest();
        this.telephone.endTest();
    }

    public void runComputer(){
        this.telephone.endTest();
        this.tablet.endTest();
    }
}
